examples/abitest/module_0/rust/src/lib.rs

```diff
@@ -2466,6 +2471,122 @@ impl FrontendNode {
         expect_matches!(result, Err(_));
         Ok(())
     }
+
+    fn test_pseudo_node_privilege(&mut self) -> TestResult {
+        // gRPC server pseudo node can be created even with Top confidentiality.
+        {
+            let top_label = oak_abi::label::confidentiality_label(oak_abi::label::top());
+            let config = oak::node_config::grpc_server(ADDITIONAL_TEST_SERVER_ADDR);
+            let (wh, rh) =
+                oak::channel_create("Test", &top_label).expect("could not create channel");
+            expect_eq!(
+                Ok(()),
+                oak::node_create("grpc_server", &config, &top_label, rh)
+            );
+            expect_eq!(Ok(()), oak::channel_close(rh.handle));
+            expect_eq!(Ok(()), oak::channel_close(wh.handle));
+        }
+
+        // HTTP server pseudo node can be created even with Top confidentiality.
+        {
+            let top_label = oak_abi::label::confidentiality_label(oak_abi::label::top());
+            let config = oak::node_config::http_server(ADDITIONAL_TEST_SERVER_ADDR);
+            let (wh, rh) =
+                oak::channel_create("Test", &top_label).expect("could not create channel");
+            expect_eq!(
+                Ok(()),
+                oak::node_create("http_server", &config, &top_label, rh)
+            );
+            expect_eq!(Ok(()), oak::channel_close(rh.handle));
+            expect_eq!(Ok(()), oak::channel_close(wh.handle));
+        }
+
+        // gRPC client pseudo node can be created with TLS endpoint confidentiality tag matching its
+        // URI authority.
+        {
+            let label = oak_abi::label::confidentiality_label(oak_abi::label::tls_endpoint_tag(
+                "localhost:7878",
+            ));
+            let config = oak::node_config::grpc_client("https://localhost:7878");
+            let (wh, rh) = oak::channel_create("Test", &label).expect("could not create channel");
+            expect_eq!(Ok(()), oak::node_create("grpc_client", &config, &label, rh));
+            expect_eq!(Ok(()), oak::channel_close(rh.handle));
+            expect_eq!(Ok(()), oak::channel_close(wh.handle));
+        }
+
+        // gRPC client pseudo node can not be created with non-matching TLS endpoint
+        // confidentiality tag.
```

Comment:
I thought that a gRPC client was allowed to declassify `domain:example.com` tags on the assumption that it's checking that the outbound TLS connection is indeed going to `example.com` ?

Reply:
Yes, that is correct. In this case it is connecting to `https://localhost:7878`, so it cannot declassify tags of `google.com`

Reply:
Ah yes, there will be a different gRPC client p-N for each domain that gets connected to.